File: mod/library/server/handle-message/exporter/_common/get-name-from-table.ts
/**
 *
 * @param tableData
 * @noSelf
 * @example
 */
const getNameFromTable = (
	tableData: (
		{
			config: never,
			nodes: {
				config: {
					object: never,
					text: any
				}
			}[]
		} |
		{
			config: never,
			nodes: {
				config: {
					object: { strings: { string: any }[] },
					text: never
				}
			}[]
		} |
		{
			config: {
				object: never,
				text: any
			},
			nodes: never
		} |
		{
			config: {
				object: { strings: { string: any }[] },
				text: never
			},
			nodes: never
		}
	)[]
) => {
	let name = "";

	for (const cell of tableData) {
		if (cell.nodes !== undefined && cell.nodes[0]?.config.object !== undefined) {
			for (const { string } of cell.nodes[0].config.object.strings) {
				name += String(string);
			}
		}
		else if (cell.nodes !== undefined && cell.nodes[0]?.config.text !== undefined) {
			name += String(cell.nodes[0].config.text);
		}
		else if (cell.config.object === undefined) {
			name += String(cell.config.text);
		}
		else {
			for (const { string } of cell.config.object.strings) {
				name += String(string);
			}
		}
	}

	return name;
};

export default getNameFromTable;

File: mod/library/server/handle-message/exporter/_common/get_name_from_table.lua
local function get_name_from_table(table)
	local name = ""
	for i, _ in ipairs(table) do
		if table[i].nodes and table[i].nodes[1].config.object then
			for _, t2 in ipairs(table[i].nodes[1].config.object.strings) do
				name = name .. tostring(t2.string)
			end
		elseif table[i].nodes then
			name = name .. tostring(table[i].nodes[1].config.text)
		elseif table[i].config.object then
			for _, t2 in ipairs(table[i].config.object.strings) do
				name = name .. tostring(t2.string)
			end
		else
			name = name .. tostring(table[i].config.text)
		end
	end
	return name
end

return get_name_from_table
